### PR TITLE
(2) Pages: Site sections rendering logic

### PR DESCRIPTION
**Pull request by rcantin-w**
## What does this change?

Relates to #11206 // For #11072 // #11280

- Checks if a page document, once fetched, has a tag that is also a valid Site Section. If so, only allow its rendering if the route matches that tag. 
- We allow orphans (`wc.org/[pageUid]`) only if the page document has no matching site section tag.
- We do not allow `/pages/` in the URL anymore, but I've kept the folder there to allow for centralised `getServerSide` and rendering logic since they all use the same one. I also added all the logic in **and this is also where we do a redirect if a user happens to be on that route (they might've had it bookmarked, for example).**

- I deleted the Visit Us/Collections "static content" from Body/ as they are only ever used on those pages, so I moved the code to the pages directly.

There is a TODO and half done code at the `getByUid`/`getBasicDocument` level that gets addressed [in this child PR](https://github.com/wellcomecollection/wellcomecollection.org/pull/11321).

## How to test

Run locally and test various routes. e.g.
- http://localhost:3000/newsletter <- Should work, valid orphan.
- http://localhost:3000/contact-us <- Should work, valid orphan.
- http://localhost:3000/pages/accessibility <- Should not work, but should be redirected to the valid a11y URL.
- http://localhost:3000/accessibility <- Should not work, not an orphan
- http://localhost:3000/visit-us/accessibility <- Should work
- http://localhost:3000/visit-us <- valid as root siteSection page.
and more.

## How can we measure success?

We're confident we can go ahead with this new logic

## Have we considered potential risks?

A lot of routes that we haven't thought about could break. We could amend those as they get flagged.

### CODE REVIEW

File: content/webapp/pages/about-us/index.tsx
Comment (davidpmccormick) [inline content/webapp/pages/about-us/index.tsx:1]: I don't understand why GitHub is showing me this as a comparison?

**Review by davidpmccormick — DISMISSED** [review]
Tested and it does the things it's supposed to. I think it makes sense 💯

**Comment by kenoir** [comment]
Is there a relationship between `ContentType` and `SiteSection`? This comes up for me because I see there is a type restriction on the `fetcher` function for content type, that perhaps avoids needing to check _after_ you've retrieved a document where it should sit.

**Comment by rcantin-w** [comment]
> Is there a relationship between ContentType and SiteSection? This comes up for me because I see there is a type restriction on the fetcher function for content type, that perhaps avoids needing to check after you've retrieved a document where it should sit.

[Addressed in this PR](https://github.com/wellcomecollection/wellcomecollection.org/pull/11283)

File: content/webapp/pages/about-us/index.tsx
Comment (rcantin-w) [reply content/webapp/pages/about-us/index.tsx:1]: Me neither, it's confused.

File: cache/edge_lambdas/src/redirects.ts [hunk 1]
```diff
@@ -15,12 +15,9 @@
  *    - User needs to remember to write it down
  *    - Print to digital transmission
  *
- * See https://app.gitbook.com/o/-LumfFcEMKx4gYXKAZTQ/s/DPDDj27NI2F2kPukWrC1/developer-handbook/urls-on-wellcomecollection.org
```
Comment (rcantin-w) [inline cache/edge_lambdas/src/redirects.ts:18]: @LaurenFBaily this page was 404ing, I don't know if you'd like to review this whole comment and add a different reference link?

**Review by gestchild — APPROVED** [review]

**Pull request merged**